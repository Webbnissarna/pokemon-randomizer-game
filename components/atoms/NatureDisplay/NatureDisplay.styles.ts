import type { ThemeUIStyleObject } from "theme-ui";

export const containerStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  alignItems: "center",
  width: "100%",
};

export const textStyle: ThemeUIStyleObject = {
  fontSize: "sm",
};

export const statContainerStyle: ThemeUIStyleObject = {
  width: "100%",
  justifyContent: "space-around",
  gap: "sm",
};

export const statStyle: ThemeUIStyleObject = {
  borderRadius: "sm",
  paddingX: "md",
  flex: 1,
  textAlign: "center",
};
