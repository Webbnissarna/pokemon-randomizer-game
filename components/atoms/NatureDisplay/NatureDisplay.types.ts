import type { Nature } from "../../../utils/Generator/pokemon.types";

export interface NatureDisplayProps {
  nature: Nature;
}
