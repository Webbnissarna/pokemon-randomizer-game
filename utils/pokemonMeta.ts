import * as CSS from "csstype";

export enum Type {
  Bug = "Bug",
  Dark = "Dark",
  Dragon = "Dragon",
  Electric = "Electric",
  Fairy = "Fairy",
  Fighting = "Fighting",
  Fire = "Fire",
  Flying = "Flying",
  Ghost = "Ghost",
  Grass = "Grass",
  Ground = "Ground",
  Ice = "Ice",
  Normal = "Normal",
  Poison = "Poison",
  Psychic = "Psychic",
  Rock = "Rock",
  Steel = "Steel",
  Water = "Water",
}

export const TypeColor: Record<Type, CSS.Property.Color> = {
  Bug: "#A8B820",
  Dark: "#6A5D55",
  Dragon: "#7038F8",
  Electric: "#F8D030",
  Fairy: "#EE99AC",
  Fighting: "#C03028",
  Fire: "#F08030",
  Flying: "#A890F0",
  Ghost: "#705898",
  Grass: "#78C850",
  Ground: "#E0C068",
  Ice: "#98D8D8",
  Normal: "#A8A878",
  Poison: "#A040A0",
  Psychic: "#F85888",
  Rock: "#B8A038",
  Steel: "#B8B8D0",
  Water: "#6890F0",
};
