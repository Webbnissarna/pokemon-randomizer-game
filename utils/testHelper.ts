import type {
  GameProp,
  GamePropGetter,
} from "./GamePropSystem/GamePropSystem.types";
import type { GeneratedPokemon } from "./Generator/generator.types";
import type { PokemonInfo, Nature, MoveInfo } from "./Generator/pokemon.types";

import Pikachu from "../public/data/pokemon/25.json";
import Pound from "../public/data/moves/1.json";
import KarateChop from "../public/data/moves/2.json";
import Fly from "../public/data/moves/19.json";
import Thunder from "../public/data/moves/87.json";
import Natures from "../public/data/naturesList.json";
import Abilities from "../public/data/abilitiesList.json";

export const MOCK_PROP: Required<GameProp> = {
  mainProp: "mainMock",
  optionalProp: "optionalMock",
};

export const MOCK_NATURE = Natures[15] as Nature;

export const mockPropGetter: GamePropGetter<GeneratedPokemon> = (_) =>
  MOCK_PROP;

export const mockPokemon: GeneratedPokemon = {
  info: Pikachu as PokemonInfo,
  nature: MOCK_NATURE,
  showdownExport: "",
  ability: Abilities[0],
  moves: [Pound, KarateChop, Fly, Thunder] as MoveInfo[],
};
