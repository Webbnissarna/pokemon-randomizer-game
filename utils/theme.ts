import { Theme, useThemeUI, ThemeUIContextValue } from "theme-ui";

const makeTheme = <T extends Theme>(t: T) => t;

const theme = makeTheme({
  colors: {
    text: "#000",
    background:
      "linear-gradient(0deg, rgba(255,172,252,1) 0%, rgba(157,214,255,1) 100%);",

    ball: {
      active: {
        top: "#ff6060",
        middle: "#606060",
        bottom: "#ffffff",
      },
      disabled: {
        top: "#4f4f4f",
        middle: "#414141",
        bottom: "#838383",
      },
    },

    positive: "#73FF67",
    positiveText: "#333333",
    negative: "#B30000",
    negativeText: "#FFFFFF",
    neutral: "#67C8FF",
    neutralText: "#000000",
    secondary: "#4E7999",
    secondaryText: "#FFFFFF",
    tertiary: "#FFFFFF",
    tertiaryText: "#000000",

    inputBackground: "#D0D0D0",
    inputText: "#3D3D3D",

    contentBackground: "#FFFFFF",
  },
  fonts: {
    body: "'Noto Sans JP', sans-serif",
    heading: "'Noto Sans JP', sans-serif",
  },
  shadows: {
    sm: "0px 2px 2px 0px rgba(0, 0, 0, 0.25)",
    md: "0px 4px 4px 0px rgba(0, 0, 0, 0.25)",

    svgFilter: {
      sm: "drop-shadow(rgba(0, 0, 0, 0.25) 0px 2px 2px)",
      md: "drop-shadow(rgba(0, 0, 0, 0.25) 0px 4px 4px)",
    },
  },
  radii: {
    sm: 5,
    md: 10,
  },
  space: {
    md: 10,
    lg: 18,
    xl: 40,
  },
  sizes: {
    sm: 30,
    md: 40,
    lg: 60,
  },
  fontSizes: {
    xxs: 9,
    xs: 10,
    sm: 12,
    md: 14,
    lg: 18,
    xl: 24,
  },

  styles: {
    root: {
      fontFamily: "body",

      h1: {
        fontSize: "xl",
      },
    },
  },
});

export default theme;
export type ThemeType = typeof theme;

interface ExactContextValue extends Omit<ThemeUIContextValue, "theme"> {
  theme: ThemeType;
}

export const useTheme = useThemeUI as unknown as () => ExactContextValue;
