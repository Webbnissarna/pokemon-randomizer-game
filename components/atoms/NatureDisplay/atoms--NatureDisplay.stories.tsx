import { Box, Flex } from "theme-ui";
import NatureDisplay from ".";
import type { Stat } from "../../../utils/Generator/pokemon.types";

interface ControlProps {
  name: string;
  increasedStat: Stat;
  decreasedStat: Stat;
}

export function Default({ name, increasedStat, decreasedStat }: ControlProps) {
  return (
    <Flex sx={{ flexDirection: "column", gap: "lg" }}>
      <h1>Custom</h1>
      <Box sx={{ width: "min(200px, 100%)" }}>
        <NatureDisplay
          nature={{
            indexNo: 0,
            name: { en: name },
            increasedStat,
            decreasedStat,
          }}
        />
      </Box>

      <h1>Nature without stats</h1>
      <Box sx={{ width: "min(200px, 100%)" }}>
        <NatureDisplay
          nature={{
            indexNo: 0,
            name: { en: "Hardy" },
            increasedStat: null,
            decreasedStat: null,
          }}
        />
      </Box>
    </Flex>
  );
}

Default.args = {
  name: "Bold",
} as ControlProps;

const statOptions = ["Attack", "Defense", "Speed", "Sp. Attack", "Sp. Defense"];

Default.argTypes = {
  increasedStat: {
    options: statOptions,
    control: { type: "select" },
  },
  decreasedStat: {
    options: statOptions,
    control: { type: "select" },
  },
};
