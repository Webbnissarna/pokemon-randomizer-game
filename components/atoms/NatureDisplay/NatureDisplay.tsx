import { Flex, Box, Text } from "theme-ui";

import { getStatColor, StatShortNames } from "../../../utils/pokemonMeta";
import {
  containerStyle,
  statContainerStyle,
  statStyle,
  textStyle,
} from "./NatureDisplay.styles";

import type { Stat } from "../../../utils/Generator/pokemon.types";
import type { NatureDisplayProps } from "./NatureDisplay.types";

export default function NatureDisplay({
  nature,
}: NatureDisplayProps): JSX.Element {
  return (
    <Flex sx={containerStyle}>
      <Text sx={textStyle}>{nature.name.en}</Text>
      <Flex sx={statContainerStyle}>
        <Box
          sx={{
            ...statStyle,
            backgroundColor: getStatColor(nature.increasedStat as Stat),
          }}
        >
          <Text sx={textStyle}>
            {nature.increasedStat
              ? `+${StatShortNames[nature.increasedStat]}`
              : "—"}
          </Text>
        </Box>
        <Box
          sx={{
            ...statStyle,
            backgroundColor: getStatColor(nature.decreasedStat as Stat),
          }}
        >
          <Text sx={textStyle}>
            {nature.decreasedStat
              ? `-${StatShortNames[nature.decreasedStat]}`
              : "—"}
          </Text>
        </Box>
      </Flex>
    </Flex>
  );
}
