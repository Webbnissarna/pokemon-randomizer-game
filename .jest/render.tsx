import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
