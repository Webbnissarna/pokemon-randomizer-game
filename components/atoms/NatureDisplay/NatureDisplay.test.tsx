import { render } from "../../../.jest/render";
import NatureDisplay from ".";
import { MOCK_NATURE } from "../../../utils/testHelper";

describe("NatureDisplay", () => {
  it("renders", () => {
    const { asFragment } = render(<NatureDisplay nature={MOCK_NATURE} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
