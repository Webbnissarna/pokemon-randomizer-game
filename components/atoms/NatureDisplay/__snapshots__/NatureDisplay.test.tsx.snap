// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatureDisplay renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-fm1h7n-Box"
  >
    <span
      class="css-3u1sqa-Box"
    >
      Naive
    </span>
    <div
      class="css-j0o3ga-Box"
    >
      <div
        class="css-1vgber0-Box"
      >
        <span
          class="css-3u1sqa-Box"
        >
          +Spd
        </span>
      </div>
      <div
        class="css-1uwxv63-Box"
      >
        <span
          class="css-3u1sqa-Box"
        >
          -Sp. Def
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
